### ISSUE TITLE
Audio drop outs from snapclients

### ISSUE THREAD

**Issue by jussbba** [badaix/snapcast#428]
Hello,

I have 3 raspberry pi's. 

First Pi is running snapserver, snapclient and mopidy. (Wifi)
Second Pi is running only snapclient. (Wifi)
Third Pi is running Home Assistant and snapclient addon. (Ethernet)

The sound from 2nd and 3rd pi is dropping out very often. (Using mainly spotify)
Should I increase the buffer or what could cause this problem? If yes, how do I change this? Snapserver config file is showing the allowed options but I am not sure where do I write down the new buffer.

I can see this in Home Assistant:

```
2018-12-14 17-10-28 [Info] Chunk: -1	-1	0	-1	500	119
2018-12-14 17-10-29 [Info] Chunk: -1	-1	-1	-1	500	119
2018-12-14 17-10-30 [Info] Chunk: 2	2	-1	-1	500	119
2018-12-14 17-10-31 [Info] Chunk: 2	2	1	-1	500	119
2018-12-14 17-10-32 [Info] Chunk: 0	0	2	-1	500	119
2018-12-14 17-10-33 [Info] Chunk: -3	-3	0	-1	500	119
2018-12-14 17-10-33 [Info] Failed to get chunk
2018-12-14 17-10-33 [Info] age > 200: 208
2018-12-14 17-10-33 [Info] sleep > bufferDuration/2: 208 > 15
2018-12-14 17-10-33 [Info] sleep > chunkDuration: 208 > 24, chunks: 1, out: 119, needed: 30
2018-12-14 17-10-33 [Info] sleep > chunkDuration: 208 > 24, chunks: 0, out: 119, needed: 30
2018-12-14 17-10-33 [Info] sleep > chunkDuration: 208 > 24, chunks: 0, out: 119, needed: 30
2018-12-14 17-10-33 [Info] sleep > chunkDuration: 197 > 24, chunks: 0, out: 119, needed: 30
2018-12-14 17-10-33 [Info] sleep > chunkDuration: 279 > 24, chunks: 0, out: 119, needed: 30
2018-12-14 17-10-33 [Info] sleep > chunkDuration: 266 > 24, chunks: 0, out: 119, needed: 30
2018-12-14 17-10-33 [Info] sleep > chunkDuration: 255 > 24, chunks: 0, out: 119, needed: 30
2018-12-14 17-10-33 [Info] sleep > chunkDuration: 250 > 24, chunks: 0, out: 119, needed: 30
2018-12-14 17-10-33 [Info] sleep > chunkDuration: 246 > 24, chunks: 0, out: 119, needed: 30
2018-12-14 17-10-33 [Info] sleep > chunkDuration: 234 > 24, chunks: 0, out: 119, needed: 30
2018-12-14 17-10-33 [Info] sleep > chunkDuration: 219 > 24, chunks: 0, out: 119, needed: 30
2018-12-14 17-10-33 [Info] sleep > chunkDuration: 204 > 24, chunks: 0, out: 119, needed: 30
2018-12-14 17-10-33 [Info] sleep > chunkDuration: 200 > 24, chunks: 0, out: 119, needed: 30
```

Edit: Adding -b 5000 to Snapserver options does not seem to make a change.
Edit: I also read something about sampleformat. Should it be 48000 or 44100? 
 
Thank you in advance!

**Comment by jussbba** [comment]
Both snapclients (2nd and 3rd pi) drop audio at the same time (Failed to get chunk)

Here is some log from the second pi:

> 2018-12-15 14-38-12 [Info] Failed to get chunk
> 2018-12-15 14-38-12 [Err] XRUN
> 2018-12-15 14-38-12 [Info] Failed to get chunk
> 2018-12-15 14-38-12 [Info] Failed to get chunk
> 2018-12-15 14-38-12 [Info] Failed to get chunk
> 2018-12-15 14-38-12 [Info] Failed to get chunk
> 2018-12-15 14-38-14 [Info] age > 200: 2024
> 2018-12-15 14-38-14 [Info] sl
> 2018-12-15 14-38-14 [Info] sleep > chunkDuration: 2024 > 26, chunks: 1, out: 0, needed: 30
> 2018-12-15 14-38-14 [Info] sleep > chunkDuration: 2025 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-14 [Info] no chunks available
> 2018-12-15 14-38-14 [Info] Failed to get chunk
> 2018-12-15 14-38-14 [Info] age > 200: 1999
> 2018-12-15 14-38-14 [Info] sleep > bufferDuration/2: 1999 > 15
> 2018-12-15 14-38-14 [Info] sleep > chunkDuration: 1999 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-14 [Info] no chunks available
> 2018-12-15 14-38-14 [Info] Failed to get chunk
> 2018-12-15 14-38-14 [Info] age > 200: 1974
> 2018-12-15 14-38-14 [Info] sleep > bufferDuration/2: 1974 > 15
> 2018-12-15 14-38-14 [Info] sleep > chunkDuration: 1974 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-14 [Info] no chunks available
> 2018-12-15 14-38-14 [Info] Failed to get chunk
> 2018-12-15 14-38-14 [Info] age > 200: 1985
> 2018-12-15 14-38-14 [Info] sleep > bufferDuration/2: 1985 > 15
> 2018-12-15 14-38-14 [Info] sleep > chunkDuration: 1985 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-14 [Info] no chunks available
> 2018-12-15 14-38-14 [Info] Failed to get chunk
> 2018-12-15 14-38-14 [Info] age > 200: 2030
> 2018-12-15 14-38-14 [Info] sleep > bufferDuration/2: 2030 > 15
> 2018-12-15 14-38-14 [Info] sleep > chunkDuration: 2030 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-14 [Info] no chunks available
> 2018-12-15 14-38-14 [Info] Failed to get chunk
> 2018-12-15 14-38-14 [Info] age > 200: 2006
> 2018-12-15 14-38-14 [Info] sleep > bufferDuration/2: 2006 > 15
> 2018-12-15 14-38-14 [Info] sleep > chunkDuration: 2006 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-14 [Info] no chunks available
> 2018-12-15 14-38-14 [Info] Failed to get chunk
> 2018-12-15 14-38-14 [Info] age > 200: 2062
> 2018-12-15 14-38-14 [Info] sleep > bufferDuration/2: 2062 > 15
> 2018-12-15 14-38-14 [Info] sleep > chunkDuration: 2062 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-14 [Info] no chunks available
> 2018-12-15 14-38-14 [Info] Failed to get chunk
> 2018-12-15 14-38-14 [Info] age > 200: 267
> 2018-12-15 14-38-14 [Info] sleep > bufferDuration/2: 267 > 15
> 2018-12-15 14-38-14 [Info] sleep > chunkDuration: 267 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-14 [Info] no chunks available
> 2018-12-15 14-38-14 [Info] Failed to get chunk
> 2018-12-15 14-38-14 [Info] age > 200: 301
> 2018-12-15 14-38-14 [Info] sleep > bufferDuration/2: 301 > 15
> 2018-12-15 14-38-14 [Info] sleep > chunkDuration: 301 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-14 [Info] no chunks available
> 2018-12-15 14-38-14 [Info] Failed to get chunk
> 2018-12-15 14-38-14 [Info] age > 200: 393
> 2018-12-15 14-38-14 [Info] sleep > bufferDuration/2: 393 > 15
> 2018-12-15 14-38-14 [Info] sleep > chunkDuration: 393 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-14 [Info] no chunks available
> 2018-12-15 14-38-14 [Info] Failed to get chunk
> 2018-12-15 14-38-14 [Info] age > 200: 368
> 2018-12-15 14-38-14 [Info] sleep > bufferDuration/2: 368 > 15
> 2018-12-15 14-38-14 [Info] sleep > chunkDuration: 368 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-14 [Info] no chunks available
> 2018-12-15 14-38-14 [Info] Failed to get chunk
> 2018-12-15 14-38-15 [Info] age > 200: 422
> 2018-12-15 14-38-15 [Info] sleep > bufferDuration/2: 422 > 15
> 2018-12-15 14-38-15 [Info] sleep > chunkDuration: 422 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-15 [Info] no chunks available
> 2018-12-15 14-38-15 [Info] Failed to get chunk
> 2018-12-15 14-38-15 [Info] age > 200: 1237
> 2018-12-15 14-38-15 [Info] sleep > bufferDuration/2:
> 2018-12-15 14-38-15 [Info] sleep > chunkDuration: 1237 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-15 [Info] sleep > chunkDuration: 1238 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-15 [Info] no chunks available
> 2018-12-15 14-38-15 [Info] Failed to get chunk
> 2018-12-15 14-38-15 [Info] age > 200: 1082
> 2018-12-15 14-38-15 [Info] sleep > bufferDuration/2: 1082 > 15
> 2018-12-15 14-38-15 [Info] sleep > chunkDuration: 1082 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-15 [Info] no chunks available
> 2018-12-15 14-38-15 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] age > 200: 1189
> 2018-12-15 14-38-16 [Info] sleep > bufferDuration/2: 1189 > 15
> 2018-12-15 14-38-16 [Info] sleep > chunkDuration: 1189 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-16 [Info] no chunks available
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] age > 200: 1313
> 2018-12-15 14-38-16 [Info] sleep > bufferDuration/2: 1313 > 15
> 2018-12-15 14-38-16 [Info] sleep > chunkDuration: 1313 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-16 [Info] no chunks available
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] age > 200: 1173
> 2018-12-15 14-38-16 [Info] sleep > bufferDuration/2: 1173 > 15
> 2018-12-15 14-38-16 [Info] sleep > chunkDuration: 1173 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-16 [Info] no chunks available
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] age > 200: 1225
> 2018-12-15 14-38-16 [Info] sleep > bufferDuration/2: 1225 > 15
> 2018-12-15 14-38-16 [Info] sleep > chunkDuration: 1225 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-16 [Info] no chunks available
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] age > 200: 1278
> 2018-12-15 14-38-16 [Info] sleep > bufferDuration/2: 1278 > 15
> 2018-12-15 14-38-16 [Info] sleep > chunkDuration: 1278 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-16 [Info] no chunks available
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] age > 200: 369
> 2018-12-15 14-38-16 [Info] sleep > bufferDuration/2: 369 > 15
> 2018-12-15 14-38-16 [Info] sleep > chunkDuration: 369 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-16 [Info] no chunks available
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] age > 200: 374
> 2018-12-15 14-38-16 [Info] sleep > bufferDuration/2: 374 > 15
> 2018-12-15 14-38-16 [Info] sleep > chunkDuration: 374 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-16 [Info] no chunks available
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] age > 200: 373
> 2018-12-15 14-38-16 [Info] sleep > bufferDuration/2: 373 > 15
> 2018-12-15 14-38-16 [Info] sleep > chunkDuration: 373 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-16 [Info] no chunks available
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] age > 200: 349
> 2018-12-15 14-38-16 [Info] sleep > bufferDuration/2: 349 > 15
> 2018-12-15 14-38-16 [Info] sleep > chunkDuration: 349 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-16 [Info] no chunks available
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] age > 200: 279
> 2018-12-15 14-38-16 [Info] sleep > bufferDuration/2: 279 > 15
> 2018-12-15 14-38-16 [Info] sleep > chunkDuration: 279 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-16 [Info] no chunks available
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] age > 200: 254
> 2018-12-15 14-38-16 [Info] sleep > bufferDuration/2: 254 > 15
> 2018-12-15 14-38-16 [Info] sleep > chunkDuration: 254 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-16 [Info] no chunks available
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] age > 200: 242
> 2018-12-15 14-38-16 [Info] sleep > bufferDuration/2: 242 > 15
> 2018-12-15 14-38-16 [Info] sleep > chunkDuration: 242 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-16 [Info] no chunks available
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] age > 200: 205
> 2018-12-15 14-38-16 [Info] sleep > bufferDuration/2: 205 > 15
> 2018-12-15 14-38-16 [Info] sleep > chunkDuration: 205 > 26, chunks: 2, out: 0, needed: 30
> 2018-12-15 14-38-16 [Info] sleep > chunkDuration: 205 > 26, chunks: 1, out: 0, needed: 30
> 2018-12-15 14-38-16 [Info] sleep > chunkDuration: 179 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-16 [Info] no chunks available
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-16 [Info] Failed to get chunk
> 2018-12-15 14-38-17 [Notice] No chunk received for 5000ms. Closing ALSA.
> 2018-12-15 14-38-17 [Info] frames: 1324
> 2018-12-15 14-38-17 [Info] age > 200: 929
> 2018-12-15 14-38-17 [Info] sleep > bufferDuration/2: 929 > 15
> 2018-12-15 14-38-17 [Info] sleep > chunkDuration: 929 > 26, chunks: 4, out: 0, needed: 30
> 2018-12-15 14-38-17 [Info] sleep > chunkDuration: 930 > 26, chunks: 3, out: 0, needed: 30
> 2018-12-15 14-38-17 [Info] sleep > chunkDuration: 930 > 26, chunks: 2, out: 0, needed: 30
> 2018-12-15 14-38-17 [Info] sleep > chunkDuration: 904 > 26, chunks: 1, out: 0, needed: 30
> 2018-12-15 14-38-17 [Info] sleep > chunkDuration: 878 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-17 [Info] no chunks available
> 2018-12-15 14-38-17 [Info] Failed to get chunk
> 2018-12-15 14-38-17 [Info] age > 200: 865
> 2018-12-15 14-38-17 [Info] sleep > bufferDuration/2: 865 > 15
> 2018-12-15 14-38-17 [Info] sleep > chunkDuration: 865 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-17 [Info] no chunks available
> 2018-12-15 14-38-17 [Info] Failed to get chunk
> 2018-12-15 14-38-17 [Notice] No chunk received for 5000ms. Closing ALSA.
> 2018-12-15 14-38-18 [Info] frames: 1324
> 2018-12-15 14-38-18 [Info] age > 200: 1196
> 2018-12-15 14-38-18 [Info] sleep > bufferDuration/2: 1196 > 15
> 2018-12-15 14-38-18 [Info] sleep > chunkDuration: 1196 > 26, chunks: 1, out: 0, needed: 30
> 2018-12-15 14-38-18 [Info] sleep > chunkDuration: 1196 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-18 [Info] no chunks available
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] age > 200: 1200
> 2018-12-15 14-38-18 [Info] sleep > bufferDuration/2: 1200 > 15
> 2018-12-15 14-38-18 [Info] sleep > chunkDuration: 1200 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-18 [Info] no chunks available
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] age > 200: 490
> 2018-12-15 14-38-18 [Info] sleep > bufferDuration/2: 490 > 15
> 2018-12-15 14-38-18 [Info] sleep > chunkDuration: 490 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-18 [Info] no chunks available
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] age > 200: 551
> 2018-12-15 14-38-18 [Info] sleep > bufferDuration/2: 551 > 15
> 2018-12-15 14-38-18 [Info] sleep > chunkDuration: 551 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-18 [Info] no chunks available
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] age > 200: 540
> 2018-12-15 14-38-18 [Info] sleep > bufferDuration/2: 540 > 15
> 2018-12-15 14-38-18 [Info] sleep > chunkDuration: 540 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-18 [Info] no chunks available
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] age > 200: 569
> 2018-12-15 14-38-18 [Info] sleep > bufferDuration/2: 569 > 15
> 2018-12-15 14-38-18 [Info] sleep > chunkDuration: 569 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-18 [Info] no chunks available
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Notice] No chunk received for 5000ms. Closing ALSA.
> 2018-12-15 14-38-18 [Info] frames: 1324
> 2018-12-15 14-38-18 [Info] age > 200: 383
> 2018-12-15 14-38-18 [Info] sleep > bufferDuration/2: 383 > 15
> 2018-12-15 14-38-18 [Info] sleep > chunkDuration: 383 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-18 [Info] no chunks available
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] age > 200: 390
> 2018-12-15 14-38-18 [Info] sleep > bufferDuration/2: 390 > 15
> 2018-12-15 14-38-18 [Info] sleep > chunkDuration: 390 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-18 [Info] no chunks available
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] age > 200: 366
> 2018-12-15 14-38-18 [Info] sleep > bufferDuration/2: 366 > 15
> 2018-12-15 14-38-18 [Info] sleep > chunkDuration: 366 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-18 [Info] no chunks available
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] age > 200: 405
> 2018-12-15 14-38-18 [Info] sleep > bufferDuration/2: 405 > 15
> 2018-12-15 14-38-18 [Info] sleep > chunkDuration: 405 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-18 [Info] no chunks available
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] age > 200: 302
> 2018-12-15 14-38-18 [Info] sleep > bufferDuration/2: 302 > 15
> 2018-12-15 14-38-18 [Info] sleep > chunkDuration: 302 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-18 [Info] no chunks available
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] age > 200: 328
> 2018-12-15 14-38-18 [Info] sleep > bufferDuration/2: 328 > 15
> 2018-12-15 14-38-18 [Info] sleep > chunkDuration: 328 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-18 [Info] no chunks available
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-18 [Info] Failed to get chunk
> 2018-12-15 14-38-19 [Notice] No chunk received for 5000ms. Closing ALSA.
> 2018-12-15 14-38-19 [Info] frames: 1324
> 2018-12-15 14-38-19 [Info] age > 200: 204
> 2018-12-15 14-38-19 [Info] sleep > bufferDuration/2: 204 > 15
> 2018-12-15 14-38-19 [Info] sleep > chunkDuration: 204 > 26, chunks: 1, out: 0, needed: 30
> 2018-12-15 14-38-19 [Info] sleep > chunkDuration: 204 > 26, chunks: 0, out: 0, needed: 30
> 2018-12-15 14-38-19 [Info] no chunks available
> 2018-12-15 14-38-19 [Info] Failed to get chunk
> 2018-12-15 14-38-19 [Info] Failed to get chunk
> 2018-12-15 14-38-19 [Info] Failed to get chunk
> 2018-12-15 14-38-19 [Info] Failed to get chunk
> 2018-12-15 14-38-19 [Info] Failed to get chunk
> 2018-12-15 14-38-19 [Info] Failed to get chunk
> 2018-12-15 14-38-19 [Info] Failed to get chunk
> 2018-12-15 14-38-19 [Info] Failed to get chunk
> 2018-12-15 14-38-19 [Info] Failed to get chunk
> 2018-12-15 14-38-19 [Info] Failed to get chunk
> 2018-12-15 14-38-19 [Info] Failed to get chunk
> 2018-12-15 14-38-19 [Info] Failed to get chunk
> 2018-12-15 14-38-19 [Info] Failed to get chunk
> 2018-12-15 14-38-19 [Info] Failed to get chunk
> 2018-12-15 14-38-19 [Info] Failed to get chunk
> 2018-12-15 14-38-19 [Info] Failed to get chunk
> 2018-12-15 14-38-19 [Info] Failed to get chunk
> 2018-12-15 14-38-19 [Info] Failed to get chunk
> 2018-12-15 14-38-19 [Info] Failed to get chunk
> 2018-12-15 14-38-19 [Info] age > 200: -218
> 2018-12-15 14-38-19 [Info] sleep < -bufferDuration/2: -218 < -15, sleep: -217
> 2018-12-15 14-38-19 [Info] sleep < -bufferDuration/2: -217 < -15, sleep: -314
> 2018-12-15 14-38-19 [Info] sleep < -bufferDuration/2: -314 < -15, sleep: -284
> 2018-12-15 14-38-19 [Info] sleep < -bufferDuration/2: -284 < -15, sleep: -254
> 2018-12-15 14-38-19 [Info] sleep < -bufferDuration/2: -254 < -15, sleep: -94
> 2018-12-15 14-38-19 [Info] sleep < -bufferDuration/2: -94 < -15, sleep: -64
> 2018-12-15 14-38-19 [Info] sleep < -bufferDuration/2: -64 < -15, sleep: -34
> 2018-12-15 14-38-19 [Info] sleep < -bufferDuration/2: -34 < -15, sleep: -4
> 2018-12-15 14-38-19 [Info] Sleep -4, age: -4, bufferDuration: 30
> 2018-12-15 14-38-19 [Info] Chunk: -44   -44     -44     -44     1       110
> 2018-12-15 14-38-19 [Info] Sleep -3, age: -3, bufferDuration: 30
> 2018-12-15 14-38-19 [Info] Sleep -2, age: -2, bufferDuration: 30
> 2018-12-15 14-38-20 [Info] Chunk: -27   -27     -27     -27     1       80
> 2018-12-15 14-38-20 [Info] Sleep -1, age: -1, bufferDuration: 30
> 2018-12-15 14-38-20 [Info] Sleep 0, age: 0, bufferDuration: 30
> 2018-12-15 14-38-20 [Info] Sleep 0
> 2018-12-15 14-38-21 [Info] Chunk: -4    -4      -4      -4      11      111

**Comment by christf** [comment]
@jussbba could you try whether my patches from #432 fix your issues?

**Comment by herrwusel** [comment]
I'm testing your patch right now and the dropouts seem less with 6500ms buffer. I need to test it in a long run because at some point audio playback just stopped for me for mopidy (not for librespot).

**Comment by christf** [comment]
interesting. Show logs please. With my patch applied I cannot see anything odd any more in the clients logs however once in a while I get audio dropouts simultaneously across two different clients. At the same time the server shows
```
2018-12-21 19-18-45 [Info] onStateChanged (default): 1
2018-12-21 19-18-45 [Err] Exception in ControlSession::reader(): read_until: Connection reset by peer
2018-12-21 19-18-48 [Info] onResync (default): 2684ms
2018-12-21 19-18-48 [Info] onStateChanged (default): 2
2018-12-21 19-18-48 [Err] Session inactive. Removing
2018-12-21 19-18-48 [Err] Error in socket shutdown: Transport endpoint is not connected
```
Could you please check for that in your server logs?

**Comment by christf** [comment]
The simultaneous audio dropouts do not occur any more with #433.

@herrwusel I am not entirely sure, yet may have seen all clients stopping to play as well. It occured once here. I do not know whether it is due to any of the patches. At the moment I am building from my next branch on https://github.com/christf/snapcast/tree/next which has our code from last week merged in.
With this I was unable to reproduce any dropouts during 5h playback in total. Please let me know how whether the patches work for you.

**Comment by herrwusel** [comment]
@christf I'm not even sure if it wasnt mopidy causing problems. I can't reproduce my problems anymore..

**Comment by christf** [comment]
in any case I am seeing this as well. The cause is a drifting clock that is eating into the buffer as described in #439

**Comment by badaix** [comment]
fixed in v0.19.0
